src/components/index.cjsx

```diff
@@ -20,15 +20,26 @@ Dashboard = React.createClass
   displayName: 'Dashboard'
   componentWillMount: -> CurrentUserStore.addChangeListener(@update)
   componentWillUnmount: -> CurrentUserStore.removeChangeListener(@update)
+  
+  contextTypes:
+    router: React.PropTypes.func
 
   update: -> @setState({})
+
   render: ->
     if CurrentUserStore.isCoursesLoaded()
       courses = CurrentUserStore.getCourses()
+      if courses.length is 1
+        roles = courses[0].roles
+        if roles.length is 1
+          singleCourseId = courses[0].id
+          if roles[0].type is 'student'
+            @context.router.transitionTo('viewStudentDashboard', {courseId: singleCourseId})
```

Comment:
maybe this logic could be in the routehandler itself so we don't break history.  because even when this transition works it will always make stepping through /dashboard/ route unstable?